sdcm/utils/topology_ops.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import traceback
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import TypeVar
+
+from sdcm.wait import wait_for
+
+LOGGER = logging.getLogger(__name__)
+BaseNode = TypeVar("BaseNode")
+BaseScyllaCluster = TypeVar("BaseScyllaCluster")
+Session = TypeVar("Session")
+
+
+class OperationState(Enum):
+    decommission = "DECOMMISSIONING"
+    shutdown = "shutdown"
+    normal = "NORMAL"
+
+
+@dataclass
+class NodeStatus:
+    ip_address: str
+    host_id: str
+    status: str
+    up: bool
+
+
+class FailedDecommissionOperationMonitoring:
+    """Monitor status of decommission operation after fail
+
+    If decommission command was failed by any reason but decommission
+    process still running, wait while it be finished and check operation status
+    """
+
+    def __init__(self, target_node: "BaseNode", verification_node: "BaseNode",
+                 timeout: int | float = 7200, expected_exception: tuple[Exception | BaseException, ...] | None = None):
+        self.timeout = timeout
+        self.target_node = target_node
+        self.db_cluster: "BaseScyllaCluster" = target_node.parent_cluster
+        self.target_node_ip = target_node.ip_address
+        expected_exception = expected_exception or set()
```

Comment:
do we need that? From my understanding it is used to skip waiting for finish node decommission if test ended (KillNemesis exception raised).
At least rename it to something that reflects that (`FailedDecommissionOperationMonitoring` that expects `KillNemesis` is weird)

Reply:
i just want to avoid annoying error event with kill nemesis, Remove that part of code

Reply:
If there's annoying error then it's a good reason to add it - my intention was to change the naming, as we don't expect any errors, rather ignoring them (or workaround possible errors due to fact of end of the test)